yt_dlp/extractor/youtube/_video.py

```diff
@@ -1812,14 +1812,18 @@ def mpd_feed(format_id, delay):
                 with lock:
                     refetch_manifest(format_id, delay)
 
-                f = next((f for f in formats if f['format_id'] == format_id), None)
+                f = next((f for f in formats if f.get('format_id') == format_id), None)
                 if not f:
                     if not is_live:
                         retry.error = f'{video_id}: Video is no longer live'
                     else:
                         retry.error = f'Cannot find refreshed manifest for format {format_id}{bug_reports_message()}'
                     continue
-                return f['manifest_url'], f['manifest_stream_number'], is_live
+
+                if not isinstance(f, dict) or not f.get('manifest_url'):
+                    break
+
+                return f.get('manifest_url'), f.get('manifest_stream_number'), is_live
```

Comment:
for a minimal diff, this should suffice yeah?

we already check `if not f` above, so we don't need to check again if it's a dict

and if `f` has a `manifest_url` key but not a `manifest_stream_number`, I'd think we'd want to error

```suggestion

                if not f.get('manifest_url'):
                    break

                return f['manifest_url'], f['manifest_stream_number'], is_live
```